Automattic/woocommerce-payments
# Make sure the WC session is initialized before checking what currency should the user see.

Fixes #3406

#### Changes proposed in this Pull Request

The MultiCurrency code stores the user's preferred currency in the WC Session. For some API requests used by the WooCommerce Blocks, the Session doesn't get initialized as part of the normal request flow, so the store's default currency is used instead. This PR changes that.

Calling `initialize_session` if the session is already initialized is basically a no-op, so there's no harm there. However, I'm not so sure about the performance implications of initializing the Session on every request. I don't see any way around that though, if we want prices to be displayed in the correct currency, we need to initialize the session.

@jessepearson Thanks for the context for this bug!

#### Testing instructions

- Have the default settings for MultiCurrency (`Automatically switch customers to their local currency`)
- Add at least another currency to the settings. For example, my store is in $ so I added €.
- Create a page of products using the "All Products" block.
- Visit the page (on an Incognito window) from a location for which the currency should change. For example, in my case, I'm in a country where € is used.
- Notice that the currency on the page matches the currency of the shop page.

-------------------


**Post merge**
